### ISSUE TITLE
Issues with installation

### ISSUE THREAD

**Issue by krzema12** [mysticfall/kotlin-react-test#1]
Hi!

When adding the dependency to my project (I use `kotlin("js")` plugin):

```
dependencies {
    testImplementation("io.github.mysticfall:kotlin-react-test:1.1.0")
}
```

I'm getting this:

```
Module <managed-automation-ui:frontend-react> has reference public react.dom/HTMLAttributes|null[0], unfortunately neither itself nor its dependencies [
	<org.jetbrains.kotlin-wrappers:kotlin-styled>
	<org.jetbrains.kotlin-wrappers:kotlin-react-dom>
	<org.jetbrains.kotlin-wrappers:kotlin-react-router-dom>
	<org.jetbrains.kotlin-wrappers:kotlin-react>
	<io.ktor:ktor-client-js>
	<org.jetbrains.kotlin-wrappers:kotlin-csstype>
	<org.jetbrains.kotlin-wrappers:kotlin-extensions>
	<org.jetbrains.kotlinx:kotlinx-html>
	<io.ktor:ktor-client-serialization>
	<io.ktor:ktor-client-json>
	<io.ktor:ktor-client-core>
	<io.ktor:ktor-http-cio>
	<io.ktor:ktor-http>
	<io.ktor:ktor-utils>
	<io.ktor:ktor-io>
	<org.jetbrains.kotlinx:kotlinx-coroutines-core>
	<org.jetbrains.kotlin-wrappers:kotlin-css>
	<org.jetbrains.kotlinx:atomicfu>
	<org.jetbrains.kotlinx:kotlinx-serialization-json>
	<org.jetbrains.kotlinx:kotlinx-serialization-core>
	<kotlin>
] contain this declaration
Please check that project configuration is correct and has required dependencies.
```

Could you advise and maybe create an example project that consumes this library?

Thanks!

PS: I created https://youtrack.jetbrains.com/issue/KT-49610 to track the topic of unit testing in React.

**Comment by krzema12** [comment]
I think it's because it's compiled only for IR backend, and I use legacy (still have to, because of no source maps in IR):
https://github.com/mysticfall/kotlin-react-test/blob/87e6171291e4a6a7b6205821e43bec2fad66bc57/build.gradle.kts#L42

**Comment by mysticfall** [comment]
Sorry for the late response. I think you are right about the source of the issue. Do you think it's better to change it to `BOTH`?

It was my understanding that they are migrating to the IR backend and I don't have a problem with source maps in my main project with that setup too.

Source maps don't seem to work for dependent libraries though but it's probably a different problem.

**Comment by krzema12** [comment]
If it's not a problem to try with `BOTH`, it would be cool. I know it would be better to migrate to IR, but it will take some time after all folks can do it, including us.

**Comment by mysticfall** [comment]
Sorry for the delay. I just published 1.2.0 version which should support both compiler backends.

Please reopen this issue if it doesn't work. Thanks!

**Comment by krzema12** [comment]
Works, thank you!
